src/ducks/sep6DepositAsset.ts

```diff
@@ -0,0 +1,313 @@
+
+      let payload = {
+        assetCode,
+        depositTypes: { ...assetInfoData.fields },
+        fields: {},
+        kycServer: "",
+        status: ActionStatus.NEEDS_INPUT,
+        token: "",
+        transferServer: tomlResponse.TRANSFER_SERVER,
+      };
+
+      if (isAuthenticationRequired) {
+        // Re-check toml for auth endpoint
+        const webAuthTomlResponse = await checkTomlForFields({
+          sepName: "SEP-6 deposit",
+          assetIssuer,
+          requiredKeys: [
+            TomlFields.WEB_AUTH_ENDPOINT,
+            TomlFields.SIGNING_KEY,
+            TomlFields.KYC_SERVER,
+          ],
+          networkUrl: networkConfig.url,
+          homeDomain,
+        });
+        log.instruction({
+          title:
+            "SEP-6 deposit is enabled, and requires authentication so we should go through SEP-10",
+        });
+        // SEP-10 start
+        const challengeTransaction = await sep10AuthStart({
+          authEndpoint: webAuthTomlResponse.WEB_AUTH_ENDPOINT,
+          serverSigningKey: webAuthTomlResponse.SIGNING_KEY,
+          publicKey,
+          homeDomain,
+        });
+
+        // SEP-10 sign
+        const signedChallengeTransaction = sep10AuthSign({
+          secretKey,
+          networkPassphrase: networkConfig.network,
+          challengeTransaction,
+        });
+
+        // SEP-10 send
+        const token = await sep10AuthSend({
+          authEndpoint: webAuthTomlResponse.WEB_AUTH_ENDPOINT,
+          signedChallengeTransaction,
+        });
+
+        // Get SEP-12 fields
+        log.instruction({
+          title: "Making GET `/customer` request for user",
+        });
+
+        const sep12Fields = await collectSep12Fields({
+          publicKey,
+          token,
+          kycServer: webAuthTomlResponse.KYC_SERVER,
+        });
+
+        payload = {
+          ...payload,
+          kycServer: webAuthTomlResponse.KYC_SERVER,
+          token,
+        };
+
+        if (sep12Fields) {
+          payload = {
+            ...payload,
+            fields: { ...payload.fields, ...sep12Fields },
+          };
+        }
+      }
+
+      return payload;
+    } catch (error) {
+      const errorMessage = getErrorMessage(error);
+
+      log.error({
+        title: "SEP-6 deposit failed",
+        body: errorMessage,
+      });
+
+      return rejectWithValue({
+        errorString: errorMessage,
+      });
+    }
+  },
+);
+
+export const submitSep6DepositFields = createAsyncThunk<
+  { status: ActionStatus; type: string },
+  {
+    depositType: AnyObject;
+    fields: AnyObject;
+  },
+  { rejectValue: RejectMessage; state: RootState }
+>(
+  "sep6DepositAsset/submitSep6DepositFields",
+  async ({ depositType, fields }, { rejectWithValue, getState }) => {
+    try {
+      const { secretKey } = accountSelector(getState());
+      const { data } = sep6DepositSelector(getState());
+      const { kycServer, token } = data;
+
+      if (Object.keys(fields).length) {
```

Comment:
if we have any customer info fields, submit them. Otherwise, just record the deposit type that the user selected

Reply:
Are we doing anything with submitted fields or the response?

Reply:
We're not - it looks like we just care about it not throwing an error. This request does respond with a generated `id`, but we don't appear to have a use for it, nor the fields that we submitted, as long as the response is a 200